pkg/vehicle/action.go

```diff
@@ -0,0 +1,70 @@
+package vehicle
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/teslamotors/vehicle-command/pkg/protocol"
+	carserver "github.com/teslamotors/vehicle-command/pkg/protocol/protobuf/carserver"
+	universal "github.com/teslamotors/vehicle-command/pkg/protocol/protobuf/universalmessage"
+	"github.com/teslamotors/vehicle-command/pkg/protocol/protobuf/vcsec"
+)
+
+// ExecuteAction executes an action on the vehicle.
+//
+// The action can be a *carserver.Action_VehicleAction or a *vcsec.UnsignedMessage.
+// Actions are created using the action package.
+func (v *Vehicle) ExecuteAction(ctx context.Context, action interface{}) error {
+	switch action := action.(type) {
+	case *carserver.Action_VehicleAction:
+		_, err := v.getCarServerResponse(ctx, action)
+		return err
+	case *vcsec.UnsignedMessage:
+		encodedPayload, err := proto.Marshal(action)
```

Comment:
Nit: Mostly for aesthetic reasons, it'd be nice if `getCarServerResponse` and `getVCSECResult` had the same signature.